minor improvement to versioned delete ops (#880)
minio/docs: source/administration/object-management/object-versioning.rst

Versioned Delete Operations
~~~~~~~~~~~~~~~~~~~~~~~~~~~

Performing a ``DELETE`` operation on a versioned object creates a 0-byte ``DeleteMarker`` as the latest version of that object.

> For objects where the latest version is a ``DeleteMarker``, clients must specify versioning flags or identifiers to ``GET/HEAD/LIST/DELETE`` operations to work on any version of that object.

Reviewer: Does this mean the client must request a specific version if the most recent is a delete marker?

I'm not sure if `to work on any version of that object` refers to any/all previous (not deleted) versions, or only a specific one.

Author: maybe "any previous version of that object" ?

Reviewer: I think that makes sense 👍🏻

> For objects where the latest version is a ``DeleteMarker``, clients must specify versioning flags or identifiers to ``GET/HEAD/LIST/DELETE`` operations to operate on any version of that object.

Author: ```suggestion
For objects where the latest version is a ``DeleteMarker``, clients must specify versioning flags or identifiers to ``GET/HEAD/LIST/DELETE`` operations to work on any version of that object.
```

The default server behavior is to omit ``DeleteMarker`` objects from consideration for unversioned operations.

MinIO relies on :ref:`Lifecycle Management expiration rules <minio-lifecycle-management-expiration>` or manual cleanup via versioned ``DELETE`` operations for removing versioned objects, including ``DeleteMarkers``. 

> MinIO does not otherwise automatically remove any versioned object.

Reviewer: Perhaps drop `otherwise`?
```suggestion
MinIO does not automatically remove any versioned object.
```

Author: Given the previous sentence, is there any way a user might somehow arrive at the conclusion that we do not support automatic removal of versioned objects? Maybe

```
MinIO relies on Lifecycle Management expiration rules for automatic cleanup *or* versioned delete operations for manual cleanup of non-current versioned objects and ``DeleteMarker`` objects.
```

?

Reviewer: Oh! I think I missed the point. Yes, you are correct. Nevermind. 😄

Author: I mean I like the new sentence better anyways so the feedback was still good

~~~~~~~~~~~~~~~~~~~~~~~~~~~

Performing a ``DELETE`` operation on a versioned object creates a 0-byte ``DeleteMarker`` as the latest version of that object.
For objects where the latest version is a ``DeleteMarker``, clients must specify versioning flags or identifiers to perform ``GET/HEAD/LIST/DELETE`` operations on a prior version of that object.
The default server behavior omits ``DeleteMarker`` objects from consideration for unversioned operations.

> MinIO relies on :ref:`Lifecycle Management expiration rules <minio-lifecycle-management-expiration>` for automatic cleanup *or* manual cleanup using versioned ``DELETE`` operations for removing non-current versioned objects and ``DeleteMarker`` objects.

Reviewer 2: ```suggestion
MinIO  can utilize :ref:`Lifecycle Management expiration rules <minio-lifecycle-management-expiration>` to automatically remove versioned objects permanently.
Otherwise, use manual ``DELETE`` operations to permanently remove non-current versioned objects or ``DeleteMarker`` objects.
```
